src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -0,0 +1,236 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static org.junit.Assert.fail;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.internal.utils.XpathUtil;
+
+public class XpathRegressionTest extends AbstractModuleTestSupport {
+
+    private static final Set<String> CHECK_NAMES = new HashSet<>(Arrays.asList(
```

Comment:
I don't see why we have a list of hardcoded check names. Is this just a list of all the checks by checkstyle in our repo?
If so, this should be using our util.
https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java#L138

Reply:
> I don't see why we have a list of hardcoded check names. Is this just a list of all the checks by checkstyle in our repo?

Yes, right. I just tried to have a complete list of checks, so that I can keep track of the remaining tasks. I went through class PackageObjectFactory, but that map is private:

https://github.com/checkstyle/checkstyle/blob/c5cc85e47ddd63c85a56c32ae26f0ad6f5310874/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L99

> https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java#L138

Thanks for your suggestion!